api.ts

```diff
@@ -605,9 +605,13 @@ export function makeFormula<
     case ValueType.String: {
       // very strange ts knows that fullDefinition.codaType is StringHintTypes but doesn't know if
       // fullDefinition is StringFormulaDefV2.
-      const {onError: _, resultType: unused, ...rest} = fullDefinition as StringFormulaDefV2<ParamDefsT>;
       const codaType = 'codaType' in fullDefinition ? fullDefinition.codaType : undefined;
+      if ('codaType' in fullDefinition) {
+        delete fullDefinition.codaType;
```

Comment:
It's a little dangerous to alter a parameter (fullDefinition is the exact object the user passed) in case the user uses it in multiple calls (very unlikely, but would result in very confusing behavior).

I wonder if we can do something like:

```
      const def: StringFormulaDefV2<ParamDefsT> = {
        ...fullDefinition,
        codaType: 'codaType' in fullDefinition ? fullDefinition.codaType : undefined,
      };
      const {onError: _, resultType: unused, codaType, ...rest} = def;
```

which would extract `codaType` from `rest` while also making `codaType` a local variable for use below.

Reply:
Sure, let me try that out. I guess our upload validation won't complain if `codaType` is present but undefined, we'll see if the problem is resolved with this approach.

 My reasoning around using `delete` was that both the entry name and value are removed from the object